src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -246,29 +245,55 @@ async function createEmscriptenModuleInstance(resourceLoader: WebAssemblyResourc
     console.error(line);
     showErrorNotification();
   };
-  const existingPreRun = moduleConfig.preRun || [];
-  const existingPostRun = moduleConfig.postRun || [];
+  const existingPreRun = moduleConfig.preRun || [] as any;
+  const existingPostRun = moduleConfig.postRun || [] as any;
   (moduleConfig as any).preloadPlugins = [];
 
   let resourcesLoaded = 0;
-  function setProgress(){
-      resourcesLoaded++;
-      const percentage = resourcesLoaded / totalResources.length * 100;
-      document.documentElement.style.setProperty('--blazor-load-percentage', `${percentage}%`);
-      document.documentElement.style.setProperty('--blazor-load-percentage-text', `"${Math.floor(percentage)}%"`);
+  function setProgress() {
+    resourcesLoaded++;
+    const percentage = resourcesLoaded / totalResources.length * 100;
+    document.documentElement.style.setProperty('--blazor-load-percentage', `${percentage}%`);
+    document.documentElement.style.setProperty('--blazor-load-percentage-text', `"${Math.floor(percentage)}%"`);
+  }
+
+  const monoToBlazorAssetTypeMap: { [key: string]: WebAssemblyBootResourceType | undefined } = {
+    'assembly': 'assembly',
+    'pdb': 'pdb',
+    'icu': 'globalization',
+    'dotnetwasm': 'dotnetwasm',
+  };
+
+  // it would not `loadResource` on types for which there is no typesMap mapping
+  const downloadResource = (request: ResourceRequest): LoadingResource | undefined => {
+    const type = monoToBlazorAssetTypeMap[request.behavior];
+    if (type !== undefined) {
+      return resourceLoader.loadResource(request.name, request.resolvedUrl!, request.hash!, type);
     }
+    return undefined;
+  };
+
+  const runtimeAssets = resourceLoader.bootConfig.resources.runtimeAssets;
```

Comment:
This is the problem for the E2E tests:

```
resourceLoader.bootConfig.resources` doesn't have a `runtimeAssets` key.  It has a `runtime` key, is that what we want?

Reply:
Looks like it needs to be in sync with 

https://github.com/dotnet/sdk/blob/4547cf714749f732b53a5c70edd269a33e75bd28/src/BlazorWasmSdk/Tasks/BootJsonData.cs#L14

according to the comment in 

https://github.com/dotnet/aspnetcore/blob/7b5da426f9bd42fa515a6b67e0bd1504eacbc6c7/src/Components/Web.JS/src/Platform/BootConfig.ts#L46-L47

and indeed that defines 

https://github.com/dotnet/sdk/blob/4547cf714749f732b53a5c70edd269a33e75bd28/src/BlazorWasmSdk/Tasks/BootJsonData.cs#L61-L64

Reply:
@lambdageek no, there was a change in the SDK (last week I think) to add this additional key.

Reply:
Ah runtimeAssets are there too. 

Maybe we just need to merge this PR with `main` to get an SDK bump?

Or maybe the E2E tests are using the sdk from `<gitroot>/.dotnet` and that one is older?

Reply:
Merging `main` wasn't enough. we probably need the SDK update PR https://github.com/dotnet/aspnetcore/pull/43028.  

I'm verifying locally...  to get past the `dotnet-install` error I'm doing something horrible to my `.nuget/packages/microsoft.dotnet.arcade.sdk/7.0.0-beta.22411.2/tools/InstallDotNetCore.targets` (commenting out the `InstallDotNetCore` task)

Reply:
Success!  We're blocked on an SDK bump

<img width="1356" alt="Screen Shot 2022-08-16 at 13 07 13" src="https://user-images.githubusercontent.com/480437/184938118-c7e2a417-479e-4e0f-933d-cb3e6a50a111.png">

Reply:
by the way this issue is the "InstallDotNetCore.targets" problem that I was having https://github.com/dotnet/arcade/issues/10423